### DISCUSSION TITLE
Issue not login and with not change web port

### DISCUSSION THREAD

**Discussion by YoshiKill** [userdocs/qbittorrent-nox-static#179] [General]
Hello, i (try to) use latest version of qbitt-nox with OpenWrt 21.02-SNAPSHOT aarch64
But i have some not understanding to me problem. 
First is that qbittorrnet is start at 8080 port and i cant change it. It's a problem because it conflict with my luci web interface, but this problem i skip changing port for luci. For qbitt i cant change port because its don't create config file. 
And main problem is that i cant log in!! 
I run qbittorrent bin as usual:
```
root@GL-MT6000:~# "/overlay/upper/usr/bin/qbittorrent" start
WebUI will be started shortly after internal preparations. Please wait...

******** Information ********
To control qBittorrent, access the WebUI at: http://localhost:8080
The WebUI administrator username is: admin
The WebUI administrator password was not set. A temporary password is provided for this session: ahT8WrYJS
You should set your own password in program preferences.

```
but when i enter Login: admin Pasw: ahT8WrYJS (or any other from restart qbitt)
its says: Invalid Username or Password. 
![Screenshot 2024-11-30 at 20-57-55 qBittorrent WebUI](https://github.com/user-attachments/assets/d28c7a1f-6e59-4f49-9a6d-581e0c93cdf7)
i try different restart with different pasword, try other browsers firefox, chrome but it's dont log in 
and there is no any config file in /.config/qBittorrent/
for change password in it or web port. 
So please help me with this problem.

**Comment by userdocs** [comment]
I have OpenWrt 23.05.5, r24106-10cc5fcd00 and it worked fine for me.

I created a new non root user and ran qbittorrent. everything worked as expected.

tbh i have no idea what this `"/overlay/upper/usr/bin/qbittorrent" start` command is supposed to be. Possibly implies a permissions issue for how you are trying to do this. Looks like a init.d service so you probably have it configured wrong, i guess.

Either you are doing something wrong or there is a bug with that version of openwrt. I cannot reproduce your issue.

The binary should simply be called with `qbittorrent-nox` if in a PATH of `/path/to/qbittorent-nox`

**Reply by YoshiKill** [reply]
> bh i have no idea what this `"/overlay/upper/usr/bin/qbittorrent" start` command is supposed to be

its my /bin/ dir with qbittorrent.bin, its just a full path to my qbittorrent.bin and start is starting service so i run it as always i run on my old router. 
how i supposed to be run it another way? 
its has all permission as chmod +x (0777) 
I recall it to qbittorrent-nox but its not the deal

> I created a new non root user and ran qbittorrent. everything worked as expected.

pls write mini faq how you do this step by step, i am noobe in owrt and unix, and don't understand how exactly do this steps

**Reply by userdocs** [reply]
```shell
opkg update
opkg install shadow
useradd -r -s /bin/ash qbittorrent
mkdir /home/qbittorrent
printf '%s\n' 'PATH="/home/qbittorrent/bin:$PATH"' > /home/qbittorrent/.profile
chown -R qbittorrent /home/qbittorrent

su -l qbittorrent

## using aarch64 as example from https://github.com/userdocs/qbittorrent-nox-static?tab=readme-ov-file#install-the-latest-release

mkdir -p ~/bin && source ~/.profile
wget -qO ~/bin/qbittorrent-nox https://github.com/userdocs/qbittorrent-nox-static/releases/latest/download/aarch64-qbittorrent-nox
chmod 700 ~/bin/qbittorrent-nox

qbittorrent-nox
```

**Reply by userdocs** [reply]
made some changes. current version should work fine.

**Reply by YoshiKill** [reply]
> ```shell
> opkg install shadow
> ```

have a problem on this first step 
```
Configuring shadow-chfn.
Configuring shadow-chage.
Configuring shadow-chgpasswd.
Collected errors:
 * check_data_file_clashes: Package shadow-chpasswd wants to install file /usr/sbin/chpasswd
        But that file is already provided by package  * busybox
 * opkg_install_cmd: Cannot install package shadow.
root@GL-MT6000:~# useradd -r -s /bin/ash qbittorrent
useradd: user 'qbittorrent' already exists
root@GL-MT6000:~# mkdir /home/qbittorrent
mkdir: can't create directory '/home/qbittorrent': No such file or directory
```
also wget is to not download any thing 
```
root@GL-MT6000:~# wget -qO ~/bin/qbittorrent-nox https://github.com/userdocs/qbittorrent-nox-static/rele
ases/latest/download/aarch64-qbittorrent-nox
```
i upload it manually via sftp  

but i cant login as i say, and don't create any config file.  On my old router with same owrt its working just fine running by root , but why on this don't i not understand.

**Reply by userdocs** [reply]
I don't know either but i'm not really going to able to support your opernwrt installation on your device.

That seems to be your primary issue, maybe try https://forum.openwrt.org

**Reply by YoshiKill** [reply]
> I don't know either but i'm not really going to able to support your opernwrt installation on your device.
> 
> That seems to be your primary issue, maybe try https://forum.openwrt.org

I understand, its can be anything, my owrt is not stock its some thing custom from Glinet. 

But likely i found solution! I just run qbittorrent-nox with keys, i forgot about it (or don't know lol) 
So with keys command it's run as well on 8090 port and finally create config folder and files qBittorrent.conf
`qbittorrent-nox --profile=/.config/qBittorrent --webui-port=8090 -d`

Now i just add to qBittorrent.conf:
`WebUI/Username=admin

WebUI/Password_ha1="@ByteArray(ARQ77eY1NUZaQsuDHbIMCA==:0WMRkYTUWVT9wVvdDtHAjU9b3b7uB8NR1Gur2hmQCvCDpm39Q+PsJRJPaCU51dEiz+dTzh8qbPsL8WkFljQYFQ==)"
`

to set login: admin pasword:adminadmin 
And its login as well.
So may be it's helps some one like me with this problem.
